ext/oci8/oci8.stub.php

```diff
@@ -2,534 +2,468 @@
  * @param array $output
  * @alias oci_fetch_all
  * @deprecated
  */
-function ocifetchstatement($statement_resource, &$output, int $skip = 0, int $maximum_rows = -1, int $flags = 0): int {}
+function ocifetchstatement($statement, &$output, int $offset = 0, int $limit = -1, int $flags = 0): int {}
 
-/**
- * @param resource $statement_resource
- */
-function oci_fetch_object($statement_resource, int $mode = PHP_OCI_ASSOC | PHP_OCI_RETURN_NULLS): stdClass|null|false {}
+/** @param resource $statement */
+function oci_fetch_object($statement, int $mode = PHP_OCI_ASSOC | PHP_OCI_RETURN_NULLS): stdClass|null|false {}
 
-/**
- * @param resource $statement_resource
- */
-function oci_fetch_row($statement_resource): array|false {}
+/** @param resource $statement */
+function oci_fetch_row($statement): array|false {}
 
-/**
- * @param resource $statement_resource
- */
-function oci_fetch_assoc($statement_resource): array|false {}
+/** @param resource $statement */
+function oci_fetch_assoc($statement): array|false {}
 
-/**
- * @param resource $statement_resource
- */
-function oci_fetch_array($statement_resource, int $mode = PHP_OCI_BOTH | PHP_OCI_RETURN_NULLS): array|false {}
+/** @param resource $statement */
+function oci_fetch_array($statement, int $mode = PHP_OCI_BOTH | PHP_OCI_RETURN_NULLS): array|false {}
 
-/**
- * @param resource $statement_resource
- */
-function oci_free_statement($statement_resource): bool {}
+/** @param resource $statement */
+function oci_free_statement($statement): bool {}
 
 /**
- * @param resource $statement_resource
+ * @param resource $statement
  * @alias oci_free_statement
  * @deprecated
  */
-function ocifreestatement($statement_resource): bool {}
+function ocifreestatement($statement): bool {}
 
 /**
- * @param resource $statement_resource
+ * @param resource $statement
  * @alias oci_free_statement
  */
-function oci_free_cursor($statement_resource): bool {}
+function oci_free_cursor($statement): bool {}
 
 /**
- * @param resource $statement_resource
+ * @param resource $statement
  * @alias oci_free_statement
  * @deprecated
  */
-function ocifreecursor($statement_resource): bool {}
+function ocifreecursor($statement): bool {}
 
-/**
- * @param resource $connection_resource
- */
-function oci_close($connection_resource): ?bool {}
+/** @param resource $connection */
+function oci_close($connection): ?bool {}
 
 /**
- * @param resource $connection_resource
+ * @param resource $connection
  * @alias oci_close
  * @deprecated
  */
-function ocilogoff($connection_resource): ?bool {}
+function ocilogoff($connection): ?bool {}
 
-/**
- * @return resource|false
- */
-function oci_new_connect(string $username, string $password, ?string $connection_string = null, string $character_set = "", int $session_mode = OCI_DEFAULT) {}
+/** @return resource|false */
+function oci_new_connect(string $username, string $password, ?string $dsn = null, string $encoding = "", int $session_mode = OCI_DEFAULT) {}
```

Comment:
Is "DSN" a general term? I thought that was a PDO-ism.

Reply:
My understanding is that it's a general database-related term. By googling for "DSN Oracle", I got multiple results, so it seems that's actually the case.

Reply:
I have a slight preference for connection_string but not enough to ask for changes if other drivers are being changed to dsn.  Python DB API uses dsn.  Go's database/sql uses dsn. The syntax the OCI8 parameter can support is getting [bigger all the time](https://download.oracle.com/ocomdocs/global/Oracle-Net-19c-Easy-Connect-Plus.pdf) and more like a 'dsn'.

Reply:
@nikic Are you sure you're okay with DSN?  :P Sorry, I couldn't resist this :D

Reply:
We're splitting hairs here, but... I checked, and currently we use `$dsn` in PDO and ODBC, and `$connection_string` in OCI and PGSQL. That seems to line up with the terminology that the upstream documentation uses. With that in mind, I'd probably just keep the `$connection_string` terminology here. But once again, splitting hairs, I think both are fine names :)

Reply:
I was about to change `$dsn` to `$connection_string`, but found out that DSN is really an important terminology in ext/pdo. There's even a pdo.dsn ini setting. Given, DSN seems like a superset of connection string (am I right?) then I think it would make sense to go with it. Especially if Oracle continues adding more capabilities to its connection string.

Reply:
Ah sorry, what I meant was to just keep the terminology DB-specific. And yeah, we definitely need to use DSN in the PDO case. So if we want to use the same everywhere, it needs to be `$dsn`.

Reply:
Alright, in the light of this (and because both of you are in favour of connection string) I changed in favour of `$connection_string` when applying this patch. :)